### ISSUE TITLE
Parameters included in set_fitparams not added to the model should trigger a warning

### ISSUE THREAD

**Issue by swiggumj** [nanograv/PINT#801]
Currently, if parameters are listed in `fitter.set_fitparams(...)` but not explicitly added to the timing model ahead of time, those params are not included in the fit. This should warn the user and suggest a fix if fits for added parameters are indeed intended.

**Comment by dlakaplan** [comment]
@swiggumj : I'll work on this.  Can you post some examples of how you've been encountering this?

**Comment by tcromartie** [comment]
As requested in #813, I'm adding that this came up in reading 'STIG' (which should be 'STIGMA') from a parfile. No warnings are given, the parameter seems to just be ignored. It might be helpful to give a warning and advise that 'STIGMA' should be used, or to just interpret it automatically.

**Comment by swiggumj** [comment]
Here's a quick example:
```
f0355 = pint.fitter.WLSFitter(t0355,m0355)
f0355.set_fitparams('RAJ','DECJ','F0','F1','DM','JUMP1')
f0355.fit_toas()
```
I've made a fitter object, run `set_fitparams` to determine parameters that are unfrozen, then I run the fit. Any parameters not included here, are frozen, however, if unknown parameters (including actual parameter names outside of existing model components) are added, they are simply ignored. For example, if I replaced the line above with:
```
f0355.set_fitparams('RAJ','DECJ','F0','F1','DM','JUMP1','CAPYBARA','PB')
```
The exact same thing will happen and no warnings are output about parameters listed that aren't actually fit (or in some cases, aren't recognized at all).

**Comment by dlakaplan** [comment]
Thanks, Joe.

Could you provide a complete example (how you generate your f0355 and m0355 objects) along with the relevant files?   If the files are too ugly to attach here, just send to me.

Also, just for completeness could you list the right way to do this?

**Comment by swiggumj** [comment]
Answered some of this offline, but for posterity...
```
source = 'J0355+28'
t0355 = toa.get_TOAs('CHIME_J0355+28_princeton.tim', usepickle=False, bipm_version='BIPM2015', ephem='DE430')
m0355 = models.get_model('J0355+28_PINT_20200929.par')
# Added to cull TOAs with large uncertainties
t0355.select(t0355.get_errors() < 1300.0*u.us)
f0355 = pint.fitter.WLSFitter(t0355,m0355)
f0355.set_fitparams('RAJ','DECJ','F0','F1','DM','JUMP1')
f0355.fit_toas()
f0355.print_summary()
```
and the par file looks like this:
```
PSR                              J0355+28
EPHEM                               DE430
UNITS                                 TDB
START              58890.9238926714457639
FINISH             59118.4808927054448726
RAJ                      3:55:22.91719108 1 0.03000022429094226922
DECJ                    28:38:40.60771679 1 1.95713138168628275615
PMRA                                  0.0
PMDEC                                 0.0
PX                                    0.0
POSEPOCH           59004.7023926881229272
F0                  2.7402575671067381387 1 8.6919332071564626984e-11
F1              -4.633787113587435083e-16 1 5.409622565523390957e-17
PEPOCH             59004.7023926881229272
PLANET_SHAPIRO                          N
NE_SW                                 4.0
SWM                                   0.0
DM                  49.401887378556114276 1 0.024948081533889613498
DM1                                   0.0
DMEPOCH            59004.7023926881229272
JUMP            -tel chime       -0.2697414083313019 1 0.0002863703797499254
```
I can make TOAs available upon request. As far as checking/adding parameters, I'm still trying to figure out the best way to do this, but @bshapiroalbert pointed me at https://github.com/nanograv/PINT/blob/master/src/pint/fitter.py: ftest function starts on line 536; relevant parameter adding starts around 623.

**Comment by dlakaplan** [comment]
OK, I have a fix at:
https://github.com/dlakaplan/PINT/tree/set_fitparams
that solves this, I think.   e.g.,
```
f0355.set_fitparams('RAJ','DECJ','F0','F1','DM','JUMP1','CAPYBARA','PB')
```
fails:
```
AttributeError: Parameter 'CAPYBARA' not present in model; cannot fit it (try to use component.add_param())
```
I think raising an error is appropriate here, rather than just a warning (to be ignored). Do others agree?

**Comment by swiggumj** [comment]
Yeah, I think that makes sense too. Looks good to me! Though I haven't actually checked it out and tried it, but I can do that shortly.

**Comment by aarchiba** [comment]
Fixed in #822 , more or less, in the sense that it tells people not to use `set_fitparams` and provides a less preposterous mechanism to do the same thing.
